imports/plugins/core/orders/client/components/invoice.js

```diff
@@ -0,0 +1,154 @@
+            </div>
+          </div>
+        }
+
+        {this.props.refunds && this.props.refunds.map((refund) => (
+          <div className="order-summary-form-group text-danger" key={refund.created} style={{ marginBottom: 15 }}>
+            <strong>Refunded on: {this.props.dateFormat(refund.created, "MM/D/YYYY")}</strong>
+            <div className="invoice-details"><strong>{formatPriceString(refund.amount)}</strong></div>
+          </div>
+        ))}
+      </div>
+    );
+  }
+
+  renderTotal() {
+    return (
+      <div className="order-summary-form-group">
+        <hr/>
+        <strong>TOTAL</strong>
+        <div className="invoice-details">
+          <strong>{formatPriceString(this.props.invoice.total)}</strong>
+        </div>
+      </div>
+    );
+  }
+
+  renderConditionalDisplay() {
+    return (
+      <div>
+        {this.props.canMakeAdjustments ?
+          <div> {this.renderTotal()} </div> :
+          <span>
+            {this.props.paymentCaptured ?
+              <div>
+                {this.renderRefundsInfo()}
+              </div>
+              :
+              <div> {this.renderTotal()} </div>
+            }
+          </span>
+        }
+      </div>
+    );
+  }
+
+  render() {
+    const invoice = this.props.invoice;
+
+    return (
+      <div>
+        <div className="order-summary-form-group">
+          <strong>Quantity Total</strong>
+          <div className="invoice-details">
+            {invoice.totalItems}
+          </div>
+        </div>
+
+        <div className="order-summary-form-group">
+          <strong><Translation defaultValue="Subtotal" i18nKey="cartSubTotals.subtotal"/></strong>
+          <div className="invoice-details">
+            {formatPriceString(invoice.subtotal)}
+          </div>
+        </div>
+
+        <div className="order-summary-form-group">
+          <strong><Translation defaultValue="Shipping" i18nKey="cartSubTotals.shipping"/></strong>
+          <div className="invoice-details">
+            {formatPriceString(invoice.shipping)}
+          </div>
+        </div>
+
+        <div className="order-summary-form-group">
+          <strong><Translation defaultValue="Tax" i18nKey="cartSubTotals.tax"/></strong>
+          <div className="invoice-details">
+            {formatPriceString(invoice.taxes)}
+          </div>
+        </div>
+
+        {this.props.discounts &&
+          <div>
+            <div className="order-summary-form-group">
+              <strong><Translation defaultValue="Discount" i18nKey="cartSubTotals.discount"/></strong>
+              <div className="invoice-details">
+                <i className="fa fa-tag fa-lg" style={{ marginRight: 2 }}/>
+                <a className="btn-link" onClick={this.props.handleClick}>Add Discount</a>
+              </div>
+            </div>
+            {this.renderDiscountForm()}
+          </div>
+        }
+        {this.renderConditionalDisplay()}
+      </div>
+    );
+  }
+}
+
+Invoice.propTypes = {
```

Comment:
This could be moved into the class as:

```
class Invoice extends Component {
    static propTypes = {}...
```

Though the way you did it is fine too, I've just been moving the to new style when I can.